students/JasneetChandok/Lesson02/series.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+"""
+a template for the series assignment
+"""
+
+
+def fibonacci(n):
+    """ compute the nth Fibonacci number """
+    if(n <= 1):
+        return n
+    else:
+        return(fibonacci(n-1) + fibonacci(n-2))
+    pass
+
+
+def lucas(n):
+    """ compute the nth Lucas number """
+    if(n == 0):
+        return 2
+    elif (n == 1):
+        return n
+    else:
+        return(lucas(n-1) + lucas(n-2))
+    pass
+
+
+def sum_series(n, n0=0, n1=1):
+    """
+    compute the nth value of a summation series.
+
+    :param n0=0: value of zeroth element in the series
+    :param n1=1: value of first element in the series
+    
+    This function should generalize the fibonacci() and the lucas(),
+    so that this function works for any first two numbers for a sum series.
+    Once generalized that way, sum_series(n, 0, 1) should be equivalent to fibonacci(n).
+    And sum_series(n, 2, 1) should be equivalent to lucas(n).
+    """
+    if n0 == False:
+        if n > 1:
+            return fibonacci(n - 1) + fibonacci(n - 2)
+        if n == 1:
+            return n1
+        elif n <= 0:
+            return n0
+    else:
+        if n > 1:
+            return lucas(n - 1) + lucas(n - 2)
```

Comment:
When you call lucas, you're always setting your n0 to 2 and n1 to 1. You will never reach the elif n == 1 or n <= 0 since the lucas functions are recursive. 

Hint: You may want to generalize the lucas function (or implement it here itself) where n0 and n1 can be anything.